pkgs/development/libraries/rustc-demangle/default.nix

```diff
@@ -0,0 +1,32 @@
+{ rustPlatform, fetchFromGitHub, rust, lib, stdenv }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rustc-demangle";
+  version = "0.1.20";
+
+  src = fetchFromGitHub {
+    owner = "alexcrichton";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-elxclyuLmr3N66s+pR4/6OU98k1oXI2wKVJtzWPY8FI=";
+  };
+
+  cargoPatches = [
+    # a patch file to add/update Cargo.lock in the source code
+    ./add-Cargo.lock.patch
```

Comment:
I think this would need instructions of how it was generated / can be regenerated, and in which cases one would need to do that.

Reply:
I generated the Cargo.lock file using `cargo generate-lockfile` from a clone of the repo since it was missing and the builder wants a lock file.

Reply:
Sounds good, please add that info to the comment so that people working on the derivation immediately know what to do (without searching through Github PR comments).

Reply:
That is the standard procedure when a lock file is missing. This wouldn't have needed documentation.

Reply:
Even if it is common, writing down (or linking to) how it's done is still very beneficial for contributors.

For example, as existing maintainer of `hotspot`, I wouldn't necessarily know how to update this new Rust dependency, so it might block me from doing future updates. A short explanation of how to do it immediately resolves that.